File: core/engineering/semantic_incident_reconstruction_engine.py
from __future__ import annotations

from typing import Any, Dict, List


MAX_INCIDENT_EVENTS = 10000


def reconstruct_semantic_incident(
    events: List[Dict[str, Any]],
) -> Dict[str, Any]:

    ordered = sorted(
        events,
        key=lambda x: (
            x.get("timestamp", 0),
            str(x.get("id")),
        ),
    )[:MAX_INCIDENT_EVENTS]

    return {
        "incident_path": ordered,
        "bounded": True,
    }

File: src/engineering/semanticIncidentReconstructionEngine.ts
/**
 * Converted from Python: core/engineering/semantic_incident_reconstruction_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export let MAX_INCIDENT_EVENTS: any = 10000;
export function reconstructSemanticIncident(events: any): any {
  var ordered: any = py.slice(py.sorted(events, {key: ((x: any) => [py.get(x, "timestamp", 0), py.toStr(py.get(x, "id"))]) as (item: any) => any}), null, MAX_INCIDENT_EVENTS);
  return {"incident_path": ordered, "bounded": true};
}
